### PR TITLE
Create an out-directory per sanitizer

### PR DESCRIPTION
**Pull request by alex**
This makes it easier to work on multiple things concurrently (most notably running a fuzzer and profiling/coverage)

### CODE REVIEW

**Comment by alex** [comment]
I tested a handful of commands by hand, but there's no automated tests so there exists the possibility of a regression

File: infra/helper.py [hunk 1]
```diff
@@ -223,9 +226,9 @@ def _get_corpus_dir(project_name=''):
   return os.path.join(BUILD_DIR, 'corpus', project_name)
 
 
-def _get_output_dir(project_name=''):
+def _get_output_dir(project_name='', sanitizer=''):
```
Comment (inferno-chromium) [inline infra/helper.py]: I think we need engine as well.

/out/libpng/libfuzzer-msan 
/out/libpng/afl-asan

**Comment by alex** [comment]
Good point

**Comment by inferno-chromium** [comment]
ah! there is atleast one isssue, we would need to repeat --sanitizer and --engine in run_* commands in documentation. Will let @oliverchang evaluate this pull request if this makes sense or keeping just /out is better.

**Comment by alex** [comment]
Yes, you need to pass the `--sanitizer` (and `--engine`) to anything that runs the binary, not just the build.

**Comment by oliverchang** [comment]
I think helper.py needs a refactor in general -- it has grown a lot than initially envisioned. 

I'm thinking a more general fix to this would be to support custom working directorys. e.g. something that looks like this (inspired by chromium's GN):

```bash
# generates some metadata files and a symlink to helper.py
$ python infra/helper.py create_config --sanitizer=blah --engine=blah /path/to/build/dir

# /path/to/build/dir remembers the sanitizer and engine config. calling helper.py frmo
# that directory automatically sets them.
$ cd /path/to/build/dir
$ python helper.py build_image # and all other commands etc
$ python helper.py build_fuzzers # stores builds in /path/to/build/dir
```

**Comment by Dor1s** [comment]
This didn't have any updates in a year, closing.

**Pull request closed without merging**